tests/common/port_toggle.py

```diff
@@ -26,19 +24,16 @@ def port_toggle(duthost, tbinfo, ports=None, wait=60, wait_after_ports_up=60, wa
         watch: Logging system state
     """
 
-    def __check_interface_state(state='up'):
-        """
-        Check interfaces status
-
-        Args:
-            state: state of DUT's interface
+    def __get_down_ports():
+        """Check interface status and return the down ports in a set
         """
+        total_down_ports = set()
         ports_down = duthost.interface_facts(up_ports=ports)['ansible_facts']['ansible_interface_link_down_ports']
-
-        if 'down' in state:
-            return len(ports_down) == len(ports)
-        else:
-            return len(ports_down) == 0
+        db_ports_down = duthost.show_interface(command='status', up_ports=ports)['ansible_facts']\
+            ['ansible_interface_link_down_ports']
```

Comment:
Does ansible_interface_link_down_ports include admin down ports? if so, this function won't behave right in the end.

Reply:
No, it only checks interfaces specified by argument `up_ports`. Admin down ports are not included in it.